src/sonic-yang-models/yang-models/sonic-macsec.yang

```diff
@@ -0,0 +1,117 @@
+module sonic-macsec {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-macsec";
+
+    prefix macsec;
+
+    description "MACsec yang Module for SONiC OS";
+
+    revision 2022-04-12 {
+        description "First Revision";
+    }
+
+
+    container sonic-macsec {
+
+        container MACSEC_PROFILE {
+
+            description "MACsec profile of config_db.json";
+
+            list MACSEC_PROFILE_LIST {
+
+                key "name";
+
+                leaf name {
+                    type string {
+                        length 1..128;
+                    }
+                }
+
+                leaf priority {
+                    type uint8;
+                    default 255;
+                }
+
+                leaf cipher_suite {
+                    type string {
+                        pattern "GCM-AES-128|GCM-AES-256|GCM-AES-XPN-128|GCM-AES-XPN-256";
+                    }
+                    default "GCM-AES-128";
+                }
+
+                leaf primary_cak {
+                    type string {
+                        pattern "[0-9a-fA-F]{32}|[0-9a-fA-F]{64}";
+                    }
+                    mandatory true;
+                }
+
+                leaf primary_ckn {
+                    type string {
+                        pattern "[0-9a-fA-F]{32}|[0-9a-fA-F]{64}";
+                    }
+                    mandatory true;
+                }
+
+                leaf fallback_cak {
+                    type string {
+                        pattern "[0-9a-fA-F]{32}|[0-9a-fA-F]{64}";
+                    }
+                }
+
+                leaf fallback_ckn {
+                    type string {
+                        pattern "[0-9a-fA-F]{32}|[0-9a-fA-F]{64}";
+                    }
+                }
+
+                must "string-length(primary_cak) = string-length(primary_ckn)";
+
+                must "string-length(fallback_cak) = string-length(fallback_ckn)";
+
+                must "string-length(fallback_cak) = string-length(primary_cak)";
+
+                must "primary_ckn != fallback_ckn";
+
+                leaf policy {
+                    type string {
+                        pattern "integrity_only|security";
+                    }
+                    default "security";
+                }
+
+                leaf enable_replay_protect {
+                    type string {
+                        pattern "true|false";
```

Comment:
https://github.com/Azure/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-feature.yang#L58
Can you use boolean_type?